dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -1626,3 +1627,112 @@ HRESULT WINAPI CFSFolder::CallBack(IShellFolder *psf, HWND hwndOwner, IDataObjec
 
     return Shell_DefaultContextMenuCallBack(this, pdtobj);
 }
+
+static HBITMAP DoLoadPicture(LPCWSTR pszFileName)
+{
+    // create stream from file
+    HRESULT hr;
+    IStream *pStream = NULL;
+    hr = SHCreateStreamOnFileEx(pszFileName, STGM_READ, FILE_ATTRIBUTE_NORMAL,
+                                FALSE, NULL, &pStream);
+    if (FAILED(hr))
+        return NULL;
+
+    // load the picture
+    HBITMAP hbm = NULL;
+    IPicture *pPicture = NULL;
+    OleLoadPicture(pStream, 0, FALSE, IID_IPicture, (LPVOID *)&pPicture);
+
+    // get the bitmap handle
+    if (pPicture)
+    {
+        pPicture->get_Handle((OLE_HANDLE *)&hbm);
+
+        // copy the bitmap handle
+        hbm = (HBITMAP)CopyImage(hbm, IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
+
+        pPicture->Release();
+    }
+
+    pStream->Release();
+
+    return hbm;
+}
+
+HRESULT WINAPI CFSFolder::GetCustomViewInfo(ULONG unknown, SFVM_CUSTOMVIEWINFO_DATA *data)
+{
+    if (data == NULL)
+    {
+        return E_POINTER;
+    }
+    if (data->cbSize != sizeof(*data))
+    {
+        // NOTE: You have to set the cbData member before SFVM_GET_CUSTOMVIEWINFO call.
+        return E_INVALIDARG;
+    }
+
+    data->hbmBack = NULL;
+    data->clrText = CLR_INVALID;
+
+    WCHAR szPath[MAX_PATH], szIniFile[MAX_PATH];
+    SHGetPathFromIDListW(pidlRoot, szPath);
+
+    // does the folder exists?
+    if (!PathIsDirectoryW(szPath))
+    {
+        return E_FAIL;
```

Comment:
Please, oh please, use a more specific value.

Reply:
For example?

Reply:
Used `E_INVALIDARG` value.